scripts/validate-workspaces.js

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env node
+
+const { readFile } = require('fs/promises');
+const { dirname, basename } = require('path');
+const { sync: glob } = require('fast-glob');
+
+/** @typedef {[path: string, manifest: Record<string, any>]} ManifestPair */
+/** @typedef {ManifestPair[]} ManifestPairs */
+
+/**
+ * @param {ManifestPairs} manifests
+ */
+function checkHaveNoDevDependencies(manifests) {
+  for (const [path, manifest] of manifests) {
+    if ('devDependencies' in manifest) {
+      throw new Error(
+        `Unexpected devDependencies in ${path}. Define devDependencies in root package.json.`,
+      );
+    }
+  }
+}
+
+/**
+ * @param {ManifestPairs} manifests
+ */
+function checkHaveCommonDependencyVersions(manifests) {
+  const versions = {};
+  for (const [path, manifest] of manifests) {
+    for (const [dependency, version] of Object.entries(manifest.dependencies || {})) {
+      if (versions[dependency] && versions[dependency] !== version) {
+        throw new Error(
+          `Inconsistent dependency version for ${dependency}@${version} in ${path}. Use a common dependency version across all workspace packages, preferring to update all to the latest version.`,
+        );
+      }
+
+      versions[dependency] = version;
+    }
+  }
+}
+
+/**
+ * @param {ManifestPairs} manifests
+ */
+function checkHaveRequiredFields(manifests) {
+  for (const [path, manifest] of manifests) {
+    ['name', 'version', 'private'].forEach((field) => {
+      if (!(field in manifest)) {
+        throw new Error(`Missing required field ${field} in ${path}`);
+      }
+    });
+  }
+}
+
+/**
+ * @param {ManifestPairs} manifests
+ */
+function checkHaveCorrectPackageName(manifests) {
+  for (const [path, manifest] of manifests) {
+    const folder = basename(dirname(path));
+    const expected = `@18f/identity-${folder}`;
+    if (manifest.name !== expected) {
+      throw new Error(`Incorrect package name ${manifest.name} in ${path}. Expected ${expected}.`);
+    }
+  }
+}
+
+/**
+ * @param {ManifestPairs} manifests
+ */
+function checkHaveCorrectVersion(manifests) {
+  for (const [path, manifest] of manifests) {
+    if (manifest.private && manifest.version !== '1.0.0') {
+      throw new Error(`Incorrect package version ${manifest.version} in ${path}. Expected 1.0.0.`);
+    }
+  }
+}
+
+/**
+ * @param {ManifestPairs} manifests
+ */
+function checkHaveNoSiblingDependencies(manifests) {
+  for (const [path, manifest] of manifests) {
+    for (const [dependency] of Object.entries(manifest.dependencies || {})) {
+      if (
+        dependency.startsWith('@18f/identity-') &&
+        manifests.some(([siblingPath]) => siblingPath === dependency.slice(14))
```

Comment:
Need to fix this, since `siblingPath` is the full relative path (`app/javascript/packages/analytics/package.json`) and `dependency.slice(14)` would just be the folder name (`analytics`).

Instead, maybe we check for a matching name in the parsed JSON manifest of other manifest pairs.

```suggestion
        manifests.some(([, { name }]) => dependency === name)
```

Reply:
Updated in 56f5afa0f.